src/Int8.mo

```diff
@@ -9,150 +10,509 @@ module {
+  ///
+  /// Int8.compare(123, 124) // => #less
+  /// ```
   public func compare(x : Int8, y : Int8) : { #less; #equal; #greater } {
     if (x < y) { #less } else if (x == y) { #equal } else { #greater }
   };
 
-  /// Returns the negation of `x`, `-x`. Traps on overflow.
+  /// Returns the negation of `x`, `-x`.
+  ///
+  /// Traps on overflow, i.e. for `neg(-2 ** 7)`.
+  ///
+  ///
+  /// Example:
+  /// ```motoko
+  /// import Int8 "mo:base/Int8";
+  ///
+  /// Int8.neg(123) // => -123
+  /// ```
   public func neg(x : Int8) : Int8 { -x };
 
-  /// Returns the sum of `x` and `y`, `x + y`. Traps on overflow.
+  /// Returns the sum of `x` and `y`, `x + y`.
+  ///
+  /// Traps on overflow/underflow.
+  ///
+  /// Example:
+  /// ```motoko
+  /// import Int8 "mo:base/Int8";
+  ///
+  /// Int8.add(100, 23) // => +123
+  /// ```
   public func add(x : Int8, y : Int8) : Int8 { x + y };
 
-  /// Returns the difference of `x` and `y`, `x - y`. Traps on underflow.
+  /// Returns the difference of `x` and `y`, `x - y`.
+  ///
+  /// Traps on overflow/underflow.
+  ///
+  /// Example:
+  /// ```motoko
+  /// import Int8 "mo:base/Int8";
+  ///
+  /// Int8.sub(123, 23) // => +100
+  /// ```
   public func sub(x : Int8, y : Int8) : Int8 { x - y };
 
-  /// Returns the product of `x` and `y`, `x * y`. Traps on overflow.
+  /// Returns the product of `x` and `y`, `x * y`.
+  ///
+  /// Traps on overflow/underflow.
+  ///
+  /// Example:
+  /// ```motoko
+  /// import Int8 "mo:base/Int8";
+  ///
+  /// Int8.mul(12, 10) // => +120
+  /// ```
   public func mul(x : Int8, y : Int8) : Int8 { x * y };
 
-  /// Returns the division of `x by y`, `x / y`.
+  /// Returns the signed integer division of `x by y`, `x / y`.
+  /// Rounds the quotient towards zero, which is the same as truncating the decimal places of the quotient.
+  ///
   /// Traps when `y` is zero.
+  ///
+  /// Example:
+  /// ```motoko
+  /// import Int8 "mo:base/Int8";
+  ///
+  /// Int8.div(123, 10) // => +12
+  /// ```
   public func div(x : Int8, y : Int8) : Int8 { x / y };
 
-  /// Returns the remainder of `x` divided by `y`, `x % y`.
+  /// Returns the remainder of the signed integer division of `x` by `y`, `x % y`,
+  /// which is defined as `x - x / y * y`.
+  ///
   /// Traps when `y` is zero.
+  ///
+  /// Example:
+  /// ```motoko
+  /// import Int8 "mo:base/Int8";
+  ///
+  /// Int8.rem(123, 10) // => +3
+  /// ```
   public func rem(x : Int8, y : Int8) : Int8 { x % y };
 
-  /// Returns `x` to the power of `y`, `x ** y`. Traps on overflow.
+  /// Returns `x` to the power of `y`, `x ** y`.
+  ///
+  /// Traps on overflow/underflow and when `y < 0 or y >= 8`.
+  ///
+  /// Example:
+  /// ```motoko
+  /// import Int8 "mo:base/Int8";
+  ///
+  /// Int8.pow(2, 6) // => +64
+  /// ```
   public func pow(x : Int8, y : Int8) : Int8 { x ** y };
 
   /// Returns the bitwise negation of `x`, `^x`.
+  /// Issue: The argument `y` is not needed and will be removed in future.
```

Comment:
Oh deah, some cut n paste happening... (by me I'm afraid)

Reply:
No problem. The changes all seem okay.